core/src/main/java/org/sonarsource/sonarlint/core/container/connected/update/objectstore/HashingPathMapper.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * SonarLint Core - Implementation
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.container.connected.update.objectstore;
+
+import java.nio.file.Path;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+/**
+ * Map strings to unique filesystem paths using their SHA1 hash.
+ */
+public class HashingPathMapper implements PathMapper<String> {
+
+  private final Path base;
+  private final int levels;
+
+  private static final String HEX_LETTERS = "0123456789abcdef";
+
+  public HashingPathMapper(Path base, int levels) {
+    if (levels < 1) {
+      throw new IllegalArgumentException("levels must be > 0");
+    }
+    if (levels > 40) {
+      throw new IllegalArgumentException("levels must be less than the length of a SHA1 hash");
+    }
+
+    this.base = base;
+    this.levels = levels;
+  }
+
+  @Override
+  public Path apply(String key) {
+    String hashedHexString = toHexString(toHash(key));
+
+    Path path = base;
+    for (int i = 0; i < levels; i++) {
+      path = path.resolve(hashedHexString.substring(i, i + 1));
+    }
+    return path.resolve(hashedHexString);
+  }
+
+  private static byte[] toHash(String key) {
+    try {
+      MessageDigest digest = MessageDigest.getInstance("SHA1");
+      digest.update(key.getBytes());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this use of "getBytes" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1943)